Documents for parallel analysis (#102)
MDAnalysis/UserGuide: doc/source/reading_and_writing.rst

Pickling
========

MDAnalysis supports pickling of most of its data structures and trajectory formats. Unsupported attributes can be found in PR `#2887 <https://github.com/MDAnalysis/mdanalysis/issues/2887>`_.

.. ipython:: python

    import pickle
    from MDAnalysis.tests.datafiles import PSF, DCD
    psf = mda.Universe(PSF, DCD)
    pickle.loads(pickle.dumps(psf))

As for :class:`MDAnalysis.core.groups.AtomGroup`, during serialization, it will be pickled with its bound

> :class:`MDAnalysis.core.universe.Universe` which means after unpickling,

Reviewer: ```suggestion
:class:`MDAnalysis.core.universe.Universe`. This means that after unpickling,
```

a new :class:`MDAnalysis.core.universe.Universe` will be created and

> be attached by the new :class:`MDAnalysis.core.groups.AtomGroup`. If the Universe is serialized

Reviewer: ```suggestion
be attached to the new :class:`MDAnalysis.core.groups.AtomGroup`. If the Universe is serialized
```

with its :class:`MDAnalysis.core.groups.AtomGroup`, they will still be bound together afterwards:

.. ipython:: python

    import pickle
    from MDAnalysis.tests.datafiles import PSF, DCD
    u = mda.Universe(PSF, DCD)
    g = u.atoms
    g_pickled = pickle.loads(pickle.dumps(g))
    print("g_pickled.universe is u: ", u is g_pickled.universe)
    g_pickled, u_pickled = pickle.loads(pickle.dumps((g, u)))
    print("g_pickled.universe is u_pickled: ",
           u_pickled is g_pickled.universe)

> If multiple :class:`MDAnalysis.core.groups.AtomGroup` are bound to the same

Reviewer: ```suggestion
If multiple :class:`MDAnalysis.core.groups.AtomGroup`\ s are bound to the same
```

> :class:`MDAnalysis.core.universe.Universe`, they will bound to the same one

Reviewer: ```suggestion
:class:`MDAnalysis.core.universe.Universe`, they will also be bound to the same one
```
